addons/account_asset/account_asset.py

```diff
@@ -1,351 +1,325 @@
+    history_ids = fields.One2many('account.asset.history', 'asset_id', string='History', readonly=True)
+    depreciation_line_ids = fields.One2many('account.asset.depreciation.line', 'asset_id', string='Depreciation Lines', readonly=True, states={'draft': [('readonly', False)], 'open': [('readonly', False)]})
+    salvage_value = fields.Float(string='Salvage Value', digits=dp.get_precision('Account'), help="It is the amount you plan to have that you cannot depreciate.", readonly=True, states={'draft': [('readonly', False)]})
+    invoice_id = fields.Many2one('account.invoice', string='Invoice', states={'draft': [('readonly', False)]}, copy=False)
+    type = fields.Selection(related="category_id.type", string='Type', required=True, default='purchase')
+
+    @api.multi
+    def unlink(self):
+        for asset in self:
+            if asset.state in ['open', 'close']:
+                raise UserError(_('You cannot delete a document is in %s state.') % (asset.state,))
+            if asset.account_move_line_ids:
+                raise UserError(_('You cannot delete a document that contains posted lines.'))
+        return super(account_asset_asset, self).unlink()
+
+    @api.model
+    def _get_period(self):
+        periods = self.env['account.period'].find()
         if periods:
             return periods[0]
-        else:
-            return False
 
-    def _get_last_depreciation_date(self, cr, uid, ids, context=None):
+    @api.multi
+    def _get_last_depreciation_date(self):
         """
         @param id: ids of a account.asset.asset objects
         @return: Returns a dictionary of the effective dates of the last depreciation entry made for given asset ids. If there isn't any, return the purchase date of this asset
         """
-        cr.execute("""
-            SELECT a.id as id, COALESCE(MAX(l.date),a.purchase_date) AS date
+        self.env.cr.execute("""
+            SELECT a.id as id, COALESCE(MAX(l.date),a.date) AS date
             FROM account_asset_asset a
             LEFT JOIN account_move_line l ON (l.asset_id = a.id)
             WHERE a.id IN %s
-            GROUP BY a.id, a.purchase_date """, (tuple(ids),))
-        return dict(cr.fetchall())
+            GROUP BY a.id, a.date """, (tuple(self.ids),))
+        result = dict(self.env.cr.fetchall())
+        return result
 
-    def _compute_board_amount(self, cr, uid, asset, i, residual_amount, amount_to_depr, undone_dotation_number, posted_depreciation_line_ids, total_days, depreciation_date, context=None):
-        #by default amount = 0
+    @api.multi
+    def _compute_board_amount(self, sequence, residual_amount, amount_to_depr, undone_dotation_number, posted_depreciation_line_ids, total_days, depreciation_date):
         amount = 0
-        if i == undone_dotation_number:
+        if sequence == undone_dotation_number:
             amount = residual_amount
         else:
-            if asset.method == 'linear':
+            if self.method == 'linear':
```

Comment:
as you are in @api.multi method, self could be a collection (orelse you need an self.ensure_one())